src/frequenz/sdk/timeseries/_base_types.py

```diff
@@ -134,6 +134,39 @@ def map(
         )
 
 
+@runtime_checkable
+class Comparable(Protocol):
+    """
+    A protocol that requires the implementation of comparison methods.
+
+    This protocol is used to ensure that types can be compared using
+    the less than or equal to (`<=`) and greater than or equal to (`>=`)
+    operators.
+    """
+
+    def __le__(self, other: Any, /) -> bool:
```

Comment:
the other could be typed as `Self`

Reply:
I did originally have that. But mypy fails in this line:
            return self.lower <= item <= self.upper
            
with the following error:
            error: Unsupported operand types for <= ("Comparable" and "_T")  [operator]
            
I do think that mypy is just wrong and this would never actually cause an issue, but I am not sure how to get around mypy.

Reply:
Yeah, it seems we are pushing the limits of the type system with this :scream:, but I think you can fix it by helping `mypy` with a `cast` here or there, as I shown in my review. I tried replacing `Any` with `Self` and I had to add one more cast to the `__contains__()` method to make it work.